IanGRap/Squish
# Refactoring managers

Contributors: Alfredo, Ian

Important Files Changed: 
* MatchManager.cs: The refactored Game manager
* BlockSpawnLocationTracker.cs: This was modified to work with the MatchManager.cs. It now takes and returns Vector3 types (previously PlayerTracker)
* Spawner.cs: This has been refactored to work with the new BlockSpawnLocationTracker and MatchManager. Most of the changes affect where the information for spawning is coming from
* CameraManager.cs: No longer adds players to the list of tracked camera objects. This is now done in the Match Manager
* UIMatchManager.cs: Winner UI functionality has been moved here. MatchManager now calls the funtions in this script to display the UI
* PlayerInput.cs : No longer destroys the inactive players. This is done in the MatchManager
* SquishDetection.cs: Now calls the MatchManager when the player is destroyed
*MatchManagerPrefab: LeftWall and RightWall have been added
* LeftWall && RightWall: the colliders have  been decreased in size slightly so that spawning blocks don't clip against the walls 
* PlayerChanges: The player AgainstWall raycast has been increased slightly due to decreasing the size of right and left wall colliders

In Depth Changes: 
* MatchManager.cs: 
    * Previously the Game Manager
    * Now keeps a list of active players - Includes functions for destroying players and removing them 
    * Holds the LeftWall and Right Wall - Includes functions for returning their positions
    * Includes public functions for returning player positions, returning wall positions and squishing players

How To Test: 
Run both sprint2 and sprint2 selections scenes. Play through the game. Make sure the game doesn't have any unexpected errors or bugs. Because this was a refactor the game should function the same as before. 
Run through the changed scripts and make sure nothing looks crazy bad.
